02_housing/2a_affordable_available_place.qmd

```diff
@@ -0,0 +1,251 @@
+
+This is the number of units that are rented to people at or below each income threshold plus the number of vacant units at that income threshold over the total number of households in each income bracket
+
+```{r}
+available_2022_final <- available_2022 %>% 
+  # join in vacancy data - note: same as in the affordable script, one place has no units that are identified as vacant
+  left_join(vacant_summed_2022, by = c("state", "place")) %>% 
+  mutate(
+    # number of affordable and available at 30 AMI per 100 households
+    share_affordable_available_30_ami_all = (at_rent30_all+Affordable30AMI_all_vacant)/Below30AMI, 
+    share_affordable_available_30_ami_renter = (at_rent30_renter+Affordable30AMI_renter_vacant)/Below30AMI, 
+    share_affordable_available_30_ami_owner = (at_rent30_owner+Affordable30AMI_owner_vacant)/Below30AMI, 
+    
+    # number of affordable and available at 50 AMI per 100 households
+    share_affordable_available_50_ami_all = (at_rent50_all+Affordable50AMI_all_vacant)/Below50AMI, 
+    share_affordable_available_50_ami_renter = (at_rent50_renter+Affordable50AMI_renter_vacant)/Below50AMI, 
+    share_affordable_available_50_ami_owner = (at_rent50_owner+Affordable50AMI_owner_vacant)/Below50AMI, 
+    
+    # number affordable and available at 80 AMI per 100 households
+    share_affordable_available_80_ami_all = (at_rent80_all+Affordable80AMI_all_vacant)/Below80AMI, 
+    share_affordable_available_80_ami_renter = (at_rent80_renter+Affordable80AMI_renter_vacant)/Below80AMI, 
+    share_affordable_available_80_ami_owner = (at_rent80_owner+Affordable80AMI_owner_vacant)/Below80AMI
+  )
+```
+
+## (5) Clean and export
+
+### (5a) subgroup file
+
+```{r}
+# combine with overall share_affordable file 
+affordable_2022_subgroup <- read_csv(here::here("02_housing/data/housing_2022_subgroups_city.csv")) %>% 
+  select(state, place, matches("quality"))
+
+# turn long for subgroup output
+available_2022_subgroup <- available_2022_final %>%
+  select(state, place, starts_with("share_affordable_available")) %>% 
+  # create year variable
+  mutate(year = 2022) %>% 
+  # seperate share_afforadable by AMI and the subgroup
+  pivot_longer(cols = c(contains("share_affordable_available")), 
+               names_to = c("available", "subgroup"),
+               names_pattern = "(.+?(?=_[^_]+$))(_[^_]+$)", # this creates two columns - "share_affordable_XXAMI" and "_owner/_renter/_all"
+               values_to = "value") %>% 
+  # pivot_wider again so that each share_affordable by AMI is it's own column with subgroups as rows
+  pivot_wider(
+    names_from = available, 
+    values_from = value
+  ) %>% 
+  # clean subgroup names and add subgroup type column 
+  # remove leading underscore and capitalize words
+  mutate(subgroup = str_remove(subgroup, "_") %>% str_to_title(),
+         subgroup_type = "tenure") %>% 
+  # join with affordable values for final file
+  left_join(affordable_2022_subgroup) %>% 
```

Comment:
This was a great catch! I merge in the `affordable_2022_subgroup` file to get the data quality variable which is the same for both the `share_affordable_` and `share_affordable_available_` values since they come from the number of households below each AMI. So this section merges in the quality flags and then renames them to make sense for this variable. I accidentally dropped the subgroup column before I merged though so it was joining wrong but fixed that.

Reply:
OK. I didn't realize that was the intention of the join. I think it's OK now.